src/components/__tests__/units/connect.test.tsx

```diff
@@ -0,0 +1,40 @@
+import { Connect } from '@/components/screens/Connect'
+import useAnalytics from '@/hooks/common/useAnalytics'
+import { render, renderHook, screen } from '@testing-library/react'
+import { MockTheme } from '@/tests-tools/theme.mock'
+import userEvent from '@testing-library/user-event'
+const onConnectMock = jest.fn()
+
+describe('Connect Screen', () => {
+	it('Page should appears', () => {
+		renderHook(() => useAnalytics())
+		render(
+			<MockTheme>
+				<Connect onConnect={onConnectMock} />
+			</MockTheme>,
+		)
+		const scan = screen.getByTestId('scan-button')
+		const input = screen.getByTestId('input-uri')
+		const connectButton = screen.getByTestId('connect-button')
+		expect(scan).toBeInTheDocument()
+		expect(input).toBeInTheDocument()
+		expect(connectButton).toBeInTheDocument()
+	})
+	it('Page should appears with Connect button and on click triggers action', async () => {
+		renderHook(() => useAnalytics())
+		const url = 'https://jestjs.io/docs/jest-object'
+		render(
+			<MockTheme>
+				<Connect onConnect={onConnectMock} mode="text" />
+			</MockTheme>,
+		)
```

Comment:
I think that you can add the custom render function even with the minimal setup. This will allow us to have it right from the start with imports. Instead of relying on testing-library import you will be able to import from tests/TestUtils for example and you will be able to update all providers at the same place.